Quotient/connectionencryptiondata_p.cpp

```diff
@@ -18,102 +18,109 @@
 using namespace Quotient;
 using namespace Quotient::_impl;
 
-Expected<PicklingKey, QKeychain::Error> setupPicklingKey(const QString& id,
-                                                         bool mock)
+void setupPicklingKey(const QString& id, bool mock, std::function<void(Expected<PicklingKey, QKeychain::Error>)> then)
```

Comment:
Can you return a `QFuture<Expected<>>` here instead of adding a function parameter? Or if you want I can do it in-branch.

Reply:
I tried doing that; I have no idea what I'm doing though and it just hangs in some future code. I'd appreciate if you could implement it